dart/dynamics/SoftBodyNode.h

```diff
@@ -361,7 +356,7 @@ class SoftBodyNode : public BodyNode
   UniqueProperties mSoftP;
 
   /// \brief Soft mesh shape belonging to this node.
-  std::shared_ptr<SoftMeshShape> mSoftShape;
+  ShapeNode* mSoftShapeNode;
```

Comment:
Won't this introduce a circular reference?! `NodePtr<>` stores a pointer to the parent `BodyNode`, which would ourselves.